File: LeetCode-December-Challenge/DailyTemperatures.java
import java.util.*;
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        Stack<Integer> myStack = new Stack<>();
        Arrays.fill(res, 0);
        for(int i = 0; i < temperatures.length; ++i){
            while(!myStack.empty() && temperatures[myStack.peek()] < temperatures[i])
                res[myStack.peek()] = i - myStack.pop();
            myStack.add(i);
        }
        return res;
    }
}
public class DailyTemperatures {
    public static void main(String[] args) {
        int[] temperatures = {73,74,75,71,69,72,76,73};
        Solution s = new Solution();
        int[] res = s.dailyTemperatures(temperatures);
        for(int x : res) System.out.print(x + " ");
    }
}

File: LeetCode-December-Challenge/DailyTemperatures.js
var dailyTemperatures = function(temperatures) {
    let res = new Array(temperatures.length).fill(0);
    let stack = [];
    for(let i = 0; i < temperatures.length; ++i){
        while(stack.length && temperatures[stack[stack.length - 1]] < temperatures[i]){
            res[stack[stack.length - 1]] = i - stack[stack.length - 1];
            stack.pop();
        }
        stack.push(i);
    }
    return res;
};
let temperatures = [73,74,75,71,69,72,76,73];
console.log(dailyTemperatures(temperatures));